src/resolve/resolved-varlink.c

```diff
@@ -551,6 +628,38 @@ int manager_varlink_init(Manager *m) {
                 return log_error_errno(r, "Failed to attach varlink connection to event loop: %m");
 
         m->varlink_server = TAKE_PTR(s);
+
+        if (m->varlink_notification_server)
+                return 0;
+
+        r = varlink_server_new(&s, VARLINK_SERVER_ACCOUNT_UID);
+        if (r < 0)
+                return log_error_errno(r, "Failed to allocate varlink server object: %m");
+
+        varlink_server_set_userdata(s, m);
+#if ENABLE_VARLINK_NOTIFICATIONS
+        r = varlink_server_bind_method_many(
+                        s,
+                        "io.systemd.ResolveNotifications.SubscribeDnsResolves",  vl_method_subscribe_dns_resolves);
+        if (r < 0)
+                return log_error_errno(r, "Failed to register varlink methods: %m");
+
+        r = varlink_server_bind_disconnect(s, vl_on_notification_disconnect);
+        if (r < 0)
+                return log_error_errno(r, "Failed to register varlink disconnect handler: %m");
+
+        r = varlink_server_listen_address(s, "/run/systemd/resolve/io.systemd.ResolveNotifications", 0660);
```

Comment:
maybe call this service io.systemd.Resolve.Monitor or so.

access should probably be 0600

Reply:
@poettering 0600 seemed a bit restrictive, are you strongly opposed to 0660?

One use case of enabling group permissions is to have another systemd service monitor notifications when using "SupplementaryGroups=" to run in the same group as systemd-resolved.

Reply:
I still have this as 0660, but if you are strongly opposed, I will change it to 0600, let me know.

Reply:
`io.systemd.Resolve.Monitor` please, 0660 is fine.

Reply:
The varlink interface is now named io.systemd.Resolve.Monitor, with method name "SubscribeDnsResolves". I've left the permissions at 0660 as agreed upon.